CMakeLists.txt

```diff
@@ -97,33 +94,30 @@ if(DJINNI_WITH_OBJC)
       ${CMAKE_INSTALL_INCLUDEDIR}/djinni/objc
   )
 
-  if (NOT DJINNI_STATIC_LIB)
-    target_link_libraries(djinni_support_lib
-      "-framework Foundation"
-    )
-  endif()
+  target_link_libraries(djinni_support_lib PUBLIC
+    "-framework Foundation"
+  )
 
 endif()
 
 target_include_directories(djinni_support_lib PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
 
 # JNI support
 option(DJINNI_WITH_JNI "Include the JNI support code in Djinni support library." off)
-option(DJINNI_JNI_WITH_MAIN "Include the default minimal JNI_OnLoad and JNI_OnUnload implementation in Djinni support library." on)
 if(DJINNI_WITH_JNI)
-  if(DJINNI_JNI_WITH_MAIN)
-    list(APPEND SRC_JNI "djinni/jni/djinni_main.cpp")
-  endif()
+  set_target_properties(djinni_support_lib PROPERTIES
+    POSITION_INDEPENDENT_CODE ON)
```

Comment:
I wonder why this was required, as the property is `ON` by default for `SHARED` and `MODULE` targets. (https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html)

Reply:
It is required when building the library as `STATIC` (which in my opinion does make sense if building for JNI). For static libraries `-fPIC` is not enabled by default. If one then tries to link the static support-lib into a shared library, it will fail.

Reply:
Thanks for the clarification!